packages/amplify-cli/src/utils/global-prefix.ts

```diff
@@ -21,3 +25,62 @@ function getYarnPrefix() {
   }
   return yarnPrefix;
 }
+
+// NOTE: This issue: https://github.com/jonschlinkert/global-prefix/issues/30 prevents us from using the global-prefix package
+function getGlobalNpmPrefix() {
+  // System-wide prefix
+  if (process.env.PREFIX) return process.env.PREFIX;
+
+  // $HOME variable
+  const home = os.homedir();
+  if (home) {
+    const prefix = tryConfigPath(path.resolve(home, '.npmrc'));
+    if (prefix) return prefix;
+  }
+
+  // Find global NPM path
+  const npm = tryWhich('npm');
+  if (npm) {
+    // Check the built-in npm config file
+    const prefix = tryConfigPath(path.resolve(npm, '..', '..', 'npmrc'));
```

Comment:
Can you elaborate a little on what this is doing? [From the docs](https://nodejs.org/api/path.html#pathresolvepaths), path.resolve it seems like there's an extraneous '..' in the array of paths.

Reply:
Sure!
What do you mean by " extraneous '..' "? I will happily make an explanation.
The goal in this line is to get the [built-in npm config](https://docs.npmjs.com/cli/v9/configuring-npm/npmrc#built-in-config-file), where you have the ini setting called `prefix`, which may have the desired value of your global `npm` path. 
According to the docs, this file can be located in the `npm` folder, which is 2 levels above `npm`'s executable (hence 2 `..`). 

If the file exists, you can find something in it's `prefix` key, which indicates the global prefix.
Then you check for the [global npm config](https://docs.npmjs.com/cli/v9/configuring-npm/npmrc#global-config-file)'s `prefix` key, and if that key doesn't exist, or the global config file itself doesn't exist, there's a fallback to your already-found `prefix`.

In a default installtion of NPM, the `npmrc` file doesn't exist. But `global-prefix` package added support for it, since it's a valid option for the user to customize according to the docs, and so did I.